pkg/api/datasources.go

```diff
@@ -93,7 +94,49 @@ func DeleteDataSourceById(c *models.ReqContext) Response {
 		return Error(403, "Cannot delete read-only data source", nil)
 	}
 
-	cmd := &models.DeleteDataSourceByIdCommand{Id: id, OrgId: c.OrgId}
+	cmd := &models.DeleteDataSourceCommand{ID: id, OrgID: c.OrgId}
+
+	err = bus.Dispatch(cmd)
+	if err != nil {
+		return Error(500, "Failed to delete datasource", err)
+	}
+
+	return Success("Data source deleted")
+}
+
+// GET /api/datasources/uid/:uid
+func GetDataSourceByUID(c *models.ReqContext) Response {
+	ds, err := getRawDataSourceByUID(c.Params(":uid"), c.OrgId)
+
+	if err != nil {
+		if errors.Is(err, models.ErrDataSourceNotFound) {
+			return Error(404, "Data source not found", nil)
+		}
+		return Error(500, "Failed to query datasources", err)
+	}
+
+	dtos := convertModelToDtos(ds)
+	return JSON(200, &dtos)
+}
+
+// DELETE /api/datasources/uid/:uid
+func DeleteDataSourceByUID(c *models.ReqContext) Response {
+	uid := c.Params(":uid")
+
+	if uid == "" {
+		return Error(400, "Missing datasource uid", nil)
+	}
+
+	ds, err := getRawDataSourceByUID(uid, c.OrgId)
+	if err != nil {
+		return Error(400, "Failed to delete datasource", nil)
+	}
+
+	if ds.ReadOnly {
+		return Error(403, "Cannot delete read-only data source", nil)
+	}
+
+	cmd := &models.DeleteDataSourceCommand{UID: uid, OrgID: c.OrgId}
```

Comment:
Nit. This part and below could be reused by DeleteDataSourceByUID and DeleteDataSourceById if extracted to `func deleteDataSource(cmd *models.DeleteDataSourceCommand) Response`. But could be fixed later if you want to get this merged fast.

Nit. If the data source is not found in `getRawDataSourceByUID` that should probably return a 404 not found instead. Same applies to `DeleteDataSourceById`.

Reply:
Fixed for 404, will save other for possible future work. Seems we have mix of responses. May want to start returning the UID (which requires the getCmd results), so will save that for possible future work focusing on api consistency.